### PR TITLE
CRM-18537 Correction of State list for Brazil

### PR DESCRIPTION
**Pull request by saurabhbatra96**
- [CRM-18537: Correction of State list for Brazil](https://issues.civicrm.org/jira/browse/CRM-18537)

### CODE REVIEW

File: CRM/Upgrade/Incremental/sql/4.7.beta8.mysql.tpl [hunk 1]
```diff
@@ -10,3 +10,5 @@ UPDATE civicrm_payment_processor pp, civicrm_payment_processor_type ppt SET pp.i
 -- CRM-17815
 {include file='../CRM/Upgrade/4.7.beta8.msg_template/civicrm_msg_template.tpl'}
 
+-- CRM-18537
+DELETE FROM civicrm_state_province WHERE name = 'Fernando de Noronha';
```
Comment (yashodha) [inline CRM/Upgrade/Incremental/sql/4.7.beta8.mysql.tpl]: I think you meant to do these changes in the latest upgrade file 4.7.8.mysql.tpl

**Comment by saurabhbatra96** [comment]
Test failures seem unrelated.

**Comment by colemanw** [comment]
jenkins, retest this please

**Comment by colemanw** [comment]
Upgrade code looks good to me.

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by JKingsnorth** [comment]
Jenkins, retest this please

**Comment by JKingsnorth** [comment]
Do we not need to handle what to do with any addresses that are using this value as well, as part of the upgrade?

**Comment by eileenmcnaughton** [comment]
good point - although the ticket suggested it was not a state when the state list was first added.... But,  you never know what people's data is like

**Comment by JKingsnorth** [comment]
ACtually, looking at the FK constraints in civicrm_address we have:
ON DELETE SET NULL for the county.

So that should handle it?

**Comment by eileenmcnaughton** [comment]
CONSTRAINT `FK_civicrm_address_state_province_id` FOREIGN KEY (`state_province_id`) REFERENCES `civicrm_state_province` (`id`) ON DELETE SET NULL,

Yep - looks good

**Comment by JKingsnorth** [comment]
That FK constraint is automatically created for custom fields that use the state list too. Hooray!

**Pull request merged**